src/Raven.Server/Documents/Replication/InterruptibleRead.cs

```diff
@@ -131,7 +131,15 @@ public void Dispose()
             _isDisposed = true;
 
             SafelyDispose(_stream); // need to dispose the current stream to abort the operation
-            SafelyDispose(_prevCall?.Result);
+            try
+            {
+                var result = _prevCall?.Result;
```

Comment:
can you describe what was fixed here?

Reply:
We are throwing here during dispose which is surprising.
What specifically happened in this case is that this task was canceled, so are disposing and instead of doing it gracefully we throwed here an exception.